modyn/storage/include/internal/grpc/storage_service_impl.hpp

```diff
@@ -520,6 +650,30 @@ class StorageServiceImpl final : public modyn::storage::Storage::Service {
     session.close();
   }
 
+  static int64_t get_number_of_samples_in_dataset_with_range(const int64_t dataset_id, soci::session& session,
+                                                             const int64_t start_timestamp = 0,
+                                                             const int64_t end_timestamp = 0) {
+    int64_t total_keys = 0;
+    std::string timestamp_filter;
+    if (start_timestamp > 0 && end_timestamp == 0) {
+      timestamp_filter = fmt::format("updated_at >= {}", start_timestamp);
+    } else if (start_timestamp == 0 && end_timestamp > 0) {
+      timestamp_filter = fmt::format("updated_at < {}", end_timestamp);
+    } else if (start_timestamp > 0 && end_timestamp > 0) {
+      timestamp_filter = fmt::format("updated_at >= {} AND updated_at < {}", start_timestamp, end_timestamp);
+    } else if (start_timestamp == 0 && end_timestamp == 0) {
+      timestamp_filter = "1 = 1";
```

Comment:
Mhm, alternatively, we could have the empty string and append AND to the filter in other cases, because this reads a bit weird at first. At least, we should add a small comment that we need some True statement in this case to keep valid SQL below

Reply:
I added a comment, because it looks neater to have `AND` outside
![image](https://github.com/eth-easl/modyn/assets/100585941/36bd72f6-3f58-44a6-b38e-33e72f51e9cf)